internal/controllers/addon/olm_reconciler.go

```diff
@@ -86,7 +86,15 @@ func (r *olmReconciler) Reconcile(ctx context.Context,
 	} else if requeueResult != resultNil {
 		return handleExit(requeueResult), nil
 	}
-	reportLastObservedAvailableCSV(addon, currentCSVKey.String())
+
+	// report addon installed condition
+	csvName := currentCSVKey.String()
+	if csvName != "" {
+		reportAddonInstalledCondition(addon, "True")
+	} else {
+		reportAddonInstalledCondition(addon, "False")
+	}
```

Comment:
Hey Ankit!
Can you make the installed condition depend on the Available condition becoming True for the first time instead?
The installed condition should only transition to True ONCE in the lifecycle of an addon with it becoming False only during uninstall - or never and the addon just vanishes.

For tracking upgrades @ashishmax31 is working on a "Upgraded" condition in #293

Reply:
@Ankit152 this is not reflective of the "installed" condition. Like Nico said, we need to set installed to true only when the addon is available for the first time(Refer the jira for more details.)

Reply:
So, when the csv comes up by the subscription fir the very first time the `csvName !=""` and `addon.Status.LastObservedAvailableCSV == ""`. This is when we will report the addon "installed" condition.
Is my thinking correct?